### DISCUSSION TITLE
Esay way to test an installation?

### DISCUSSION THREAD

**Discussion by DonKracho** [sle118/squeezelite-esp32#350] [Q&A] [answered]
I have got a ESP32-WROVER-DEV camera clone and wanted to test the sueezelite functionality without applying any additional hardware. The module claims to have an ESP32-WROVER-E and 64MBit (8MB) of PSRAM. Using the Web installer with the Generic/I2S image ends up in a boot loop. Using the ESP Muse Deluxe image (Muse-16 2.1634) boots up and I can access the setup web page.

I configued it to use bluetooth as output. and set a the bluetooth device name as shown on my phone. But nor does the ESP connect to the BT speaker neither does it get listed as player in my LMS instances (ONAP-Docker and Raspberry Pi 3b)

Am I missing something? Should it work this way?

**Comment by philippe44** [comment]
The I2S should work, unless the PSRAM or Flash have a special timing. What do you have as logs?

**Reply by DonKracho** [reply]
I have not a DAC lying arround therefore there is noting connected to I2S. I wanted to test the installation with the bare module. I hope to get some ESP32 Audio Kits next week from China. These are prove to work with ES32-squeezelight. May be I should wait for them.

LMS is detected but LMS does not see the player.

[nvs_config_squeezelite-752520_1695536641363json.txt](https://github.com/sle118/squeezelite-esp32/files/13273928/nvs_config_squeezelite-752520_1695536641363json.txt)

console output:
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:6424
load:0x40078000,len:15632
load:0x40080400,len:4348
entry 0x40080688
␛[0;32mI (27) boot: ESP-IDF v4.3.5-dirty 2nd stage bootloader␛[0m
␛[0;32mI (27) boot: compile time 03:34:47␛[0m
␛[0;32mI (27) boot: chip revision: v3.0␛[0m
␛[0;32mI (31) qio_mode: Enabling default flash chip QIO␛[0m
␛[0;32mI (36) boot.esp32: SPI Speed      : 80MHz␛[0m
␛[0;32mI (41) boot.esp32: SPI Mode       : QIO␛[0m
␛[0;32mI (45) boot.esp32: SPI Flash Size : 4MB␛[0m
␛[0;32mI (50) boot: Enabling RNG early entropy source...␛[0m
␛[0;32mI (55) boot: Partition Table:␛[0m
␛[0;32mI (59) boot: ## Label            Usage          Type ST Offset   Length␛[0m
␛[0;32mI (66) boot:  0 nvs              WiFi data        01 02 00009000 00004000␛[0m
␛[0;32mI (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000␛[0m
␛[0;32mI (81) boot:  2 phy_init         RF data          01 01 0000f000 00001000␛[0m
␛[0;32mI (88) boot:  3 recovery         factory app      00 00 00010000 00140000␛[0m
␛[0;32mI (96) boot:  4 ota_0            OTA app          00 10 00150000 002a0000␛[0m
␛[0;32mI (103) boot:  5 settings         WiFi data        01 02 003f0000 00010000␛[0m
␛[0;32mI (111) boot: End of partition table␛[0m
␛[0;32mI (115) boot: Defaulting to factory image␛[0m
␛[0;32mI (120) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=53840h (342080) map␛[0m
␛[0;32mI (223) esp_image: segment 1: paddr=00063868 vaddr=3ffbdb60 size=04848h ( 18504) load␛[0m
␛[0;32mI (229) esp_image: segment 2: paddr=000680b8 vaddr=40080000 size=07f60h ( 32608) load␛[0m
␛[0;32mI (239) esp_image: segment 3: paddr=00070020 vaddr=400d0020 size=d1444h (857156) map␛[0m
␛[0;32mI (476) esp_image: segment 4: paddr=0014146c vaddr=40087f60 size=0caech ( 51948) load␛[0m
␛[0;32mI (493) esp_image: segment 5: paddr=0014df60 vaddr=400c0000 size=00068h (   104) load␛[0m
␛[0;32mI (504) boot: Loaded app from partition at offset 0x10000␛[0m
␛[0;32mI (504) boot: Disabling RNG early entropy source...␛[0m
␛[0;32mI (515) psram: This chip is ESP32-D0WD␛[0m
␛[0;32mI (515) spiram: Found 64MBit SPI RAM device␛[0m
␛[0;32mI (516) spiram: SPI RAM mode: flash 80m sram 80m␛[0m
␛[0;32mI (519) spiram: PSRAM initialized, cache is in low/high (2-core) mode.␛[0m
␛[0;32mI (526) cpu_start: Pro cpu up.␛[0m
␛[0;32mI (530) cpu_start: Starting app cpu, entry point is 0x400812a0␛[0m
␛[0;32mI (523) cpu_start: App cpu up.␛[0m
␛[0;32mI (1035) spiram: SPI SRAM memory test OK␛[0m
␛[0;32mI (1045) cpu_start: Pro cpu start user code␛[0m
␛[0;32mI (1045) cpu_start: cpu freq: 240000000␛[0m
␛[0;32mI (1045) cpu_start: Application information:␛[0m
␛[0;32mI (1048) cpu_start: Project name:     SqueezeAMP␛[0m
␛[0;32mI (1053) cpu_start: App version:      SqueezeAmp-1634␛[0m
␛[0;32mI (1059) cpu_start: Compile time:     Oct 29 2023 03:36:28␛[0m
␛[0;32mI (1065) cpu_start: ELF file SHA256:  7e8197f3afd62980...␛[0m
␛[0;32mI (1071) cpu_start: ESP-IDF:          v4.3.5-dirty␛[0m
␛[0;32mI (1076) cpu_start: Min chip rev:     v1.0␛[0m
␛[0;32mI (1081) cpu_start: Max chip rev:     v3.99 ␛[0m
␛[0;32mI (1086) cpu_start: Chip rev:         v3.0␛[0m
␛[0;32mI (1091) heap_init: Initializing. RAM available for dynamic allocation:␛[0m
␛[0;32mI (1098) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM␛[0m
␛[0;32mI (1104) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM␛[0m
␛[0;32mI (1111) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM␛[0m
␛[0;32mI (1117) heap_init: At 3FFC47E0 len 0001B820 (110 KiB): DRAM␛[0m
␛[0;32mI (1123) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM␛[0m
␛[0;32mI (1130) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM␛[0m
␛[0;32mI (1136) heap_init: At 40094A4C len 0000B5B4 (45 KiB): IRAM␛[0m
␛[0;32mI (1142) spiram: Adding pool of 4059K of external SPI memory to heap allocator␛[0m
␛[0;32mI (1151) spi_flash: detected chip: generic␛[0m
␛[0;32mI (1155) spi_flash: flash io: qio␛[0m
␛[0;32mI (1160) cpu_start: Starting scheduler on PRO CPU.␛[0m
␛[0;32mI (0) cpu_start: Starting scheduler on APP CPU.␛[0m
␛[0;32mI (1170) spiram: Reserving pool of 64K of internal memory for DMA/internal allocations␛[0m
␛[0;32mI (1180) esp_app_main: Reset reason is: 3␛[0m
␛[0;32mI (1180) esp_app_main: Recovery Reboot counter=1
␛[0m
␛[0;32mI (1190) esp_app_main: Starting app_main␛[0m
␛[0;32mI (1190) nvs_utilities: Initializing flash nvs ␛[0m
␛[0;32mI (1210) nvs_utilities: Initializing nvs partition settings␛[0m
␛[0;32mI (1260) esp_app_main: Setting up telnet.␛[0m
␛[0;32mI (1260) telnet: ***Redirecting log output to telnet␛[0m
␛[0;32mI (1260) esp_app_main: Setting up config subsystem.␛[0m
␛[0;33mW (1330) nvs_utilities: Configuration memory usage.  Heap internal:261755 (min:261379) (used:15372) external:4111887 (min:4104415) (used:0)␛[0m
␛[0;32mI (1330) esp_app_main: Registering default values␛[0m
␛[0;32mI (1340) system_api: Base MAC address is not set␛[0m
␛[0;32mI (1340) system_api: read default base MAC address from EFUSE␛[0m
␛[0;33mW (1350) config: Waiting for config commit ...␛[0m
␛[0;32mI (2330) config: configuration has some uncommitted entries␛[0m
␛[0;32mI (2330) config: Committing configuration to nvs. Locking config object.␛[0m
␛[0;32mI (2330) config: Done Committing configuration to nvs.␛[0m
␛[0;32mI (2330) config: Config committed!␛[0m
␛[0;32mI (2340) esp_app_main: Configuring services␛[0m
␛[0;32mI (2340) services: Configuring I2C sda:-1 scl:-1 port:1 speed:400000␛[0m
␛[0;33mW (2350) services: no I2C configured␛[0m
␛[0;32mI (2350) services: Configuring SPI mosi:-1 miso:-1 clk:-1 host:1 dc:-1␛[0m
␛[0;33mW (2360) services: no SPI configured␛[0m
␛[0;32mI (2370) led: Configuring LEDs green:12 (on:0 rmt:-1 -1% ), red:13 (on:0 rmt:-1 -1% )␛[0m
␛[0;32mI (2370) battery: Battery measure channel: 7, scale 20.240000, atten 0, cells 2, avg 0.91V␛[0m
␛[0;32mI (2380) monitor: Initializing monitoring␛[0m
␛[0;32mI (2390) monitor: Adding jack (low) detection GPIO 34␛[0m
␛[0;32mI (2390) buttons: Creating button using GPIO 34, type 0, pull-up/down 0, long press 0 shifter -1␛[0m
␛[0;32mI (2400) monitor: Adding speaker fault (low) detection GPIO 2␛[0m
␛[0;32mI (2400) buttons: starting button tasks␛[0m
␛[0;32mI (2410) buttons: Creating button using GPIO 2, type 0, pull-up/down 0, long press 0 shifter -1␛[0m
␛[0;32mI (2420) monitor: Heap internal:261863 (min:261255) external:4094975 (min:4094959) dma:216151 (min:215559)␛[0m
␛[0;32mI (2430) esp_app_main: Initializing display␛[0m
␛[0;32mI (2440) display: Trying to configure display with N/A␛[0m
␛[0;33mW (2440) display: No display driver␛[0m
␛[0;32mI (2450) esp_app_main: Initializing led_vu␛[0m
␛[0;32mI (2450) led_vu: led_vu configuration invalid␛[0m
␛[0;32mI (2460) esp_app_main: Checking for update url␛␛[0;33mW (2470) config: Value not found for key apdelay␛[0m
␛[0;32mI (2470) uart: queue free spaces: 3␛[0m
␛[0;32mI (2470) network_handlers: Network manager configuration: polling max 10176, polling min 15000, ap delay 20000, dhcp timeout 8000, eth timeout 8000␛[0m
␛[0;32mI (2500) http_server: Initializing HTTP Server␛[0m
␛[0;32mI (2510) network_handlers: Running recovery. Skipping ethernet, starting WiFi␛[0m
I (2520) wifi:wifi driver task: 3ffca488, prio:23, stack:6656, core=0
I (2520) wifi:wifi firmware version: 9ec65ef
I (2520) wifi:wifi certification version: v7.0
I (2520) wifi:config NVS flash: enabled
I (2530) wifi:config nano formating: disabled
I (2530) wifi:Init data frame dynamic rx buffer num: 40
I (2530) wifi:Init management frame dynamic rx buffer num: 40
I (2540) wifi:Init management short buffer num: 32
I (2540) wifi:Init static tx buffer num: 12
I (2550) wifi:Init tx cache buffer num: 32

I (2560) wifi:Init static rx buffer size: 1600
I (2560) wifi:Init static rx buffer num: 12
I (2560) wifi:Init dynamic rx buffer num: 40****************************************************************
RECOVERY APPLICATION
This mode is used to flash Squeezelite into the OTA partition
****


␛[0;32mI (2580) wifi_init: tcpip mbox: 32␛[0m
␛[0;32mI (2580) wifi_init: udp mbox: 32␛[0m
␛[0;32mI (2580) wifi_init: tcp mbox: 32␛[0m
Type 'help' to get the list of commands.
Use UP/DOWN arrows to navigate through command history.
Press TAB when typing command name to auto-complete.

␛[0;32mI (2590) wifi_init: tcp tx win: 8192␛[0m
␛[0;32mI (2610) wifi_init: tcp rx win: 32768␛[0m


␛[0;32mI (2620) wifi_init: tcp mss: 1440␛[0m
␛[0;32mI (2620) wifi_init: WiFi/LWIP prefer SPIRAM␛[0m
␛[0;32mI (2620) network_wifi: Starting wifi␛[0m

****************************
Your terminal application does not support escape sequences.
Line editing and history features are disabled.
On Windows, try using Putty instead.
****************************
␛[0;32mI (2630) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07␛[0m
recovery-squeezelite-esp32>I (2740) wifi:mode : sta (d4:d4:da:5a:fd:28)
I (2740) wifi:enable tsf
␛[0;32mI (2740) network_handlers: Existing wifi config found. Attempting to connect.␛[0m
␛[0;32mI (2740) network: Starting DHCP client␛[0m
␛[0;32mI (2760) network_wifi: Wifi Connecting to Crypton...␛[0m
I (5160) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:1
I (5720) wifi:state: init -> auth (b0)
I (5730) wifi:state: auth -> assoc (0)
I (5740) wifi:state: assoc -> run (10)
I (5770) wifi:connected with Crypton, aid = 9, channel 1, BW20, bssid = 44:4e:6d:68:7d:fb
I (5770) wifi:security: WPA2-PSK, phy: bgn, rssi: -58
I (5770) wifi:pm start, type: 1

␛[0;32mI (5790) network: Got an IP address from interface Wifi. IP=192.168.0.128, Gateway=192.168.0.1, NetMask=255.255.254.0, Address was changed␛[0m
␛[0;32mI (5800) esp_netif_handlers: sta ip: 192.168.0.128, mask: 255.255.254.0, gw: 192.168.0.1␛[0m
␛[0;32mI (5800) esp_app_main: Network connected and mDNS initialized with squeezelite-5afd28␛[0m
␛[0;32mI (5820) squeezelite-ota: sending LMS discovery for OTA Update␛[0m
␛[0;32mI (5830) squeezelite-ota: got response from: 192.168.0.250:3483 - EJSON␄9001CLIP␄9090␛[0m
␛[0;32mI (5830) network_status: LMS IP: 192.168.0.250, hport: 9001, cport: 9090␛[0m
I (5840) wifi:AP's beacon interval = 102400 us, DTIM period = 1
␛[0;32mI (8060) httpd_handlers: serving /commands.json to peer 192.168.1.3 port 35024␛[0m
␛[0;32mI (8280) httpd_handlers: serving /messages.json to peer 192.168.1.3 port 35280␛[0m
␛[0;32mI (8350) httpd_handlers: serving /status.json to peer 192.168.1.3 port 34256␛[0m
␛[0;32mI (11500) monitor: Heap internal:137495 (min:136579) external:3995415 (min:3965911) dma:91783 (min:90883)␛[0m
␛[0;32mI (21600) monitor: Heap internal:137271 (min:136579) external:3994623 (min:3965911) dma:91559 (min:90883)␛[0m
␛[0;32mI (21600) monitor:             IDLE (1) 99% s:  948     pseudo_idle (0)  0% s: 3316            IDLE (1) 97% s:  948␛[0m
␛[0;32mI (21620) monitor:              tiT (2)  0% s: 1884         Tmr Svc (2)  0% s: 1200         console (2)  0% s: 2828␛[0m
␛[0;32mI (21640) monitor:             ipc1 (2)  0% s:  464            ipc0 (2)  0% s:  468         sys_evt (2)  0% s:  808␛[0m
␛[0;32mI (21640) monitor:             mdns (2)  0% s: 2972         network (2)  0% s:  800           httpd (2)  0% s: 1140␛[0m
␛[0;32mI (21650) monitor:        esp_timer (3)  0% s: 3372            wifi (2)  1% s: 3196         buttons (2)  0% s: 1548␛[0m
␛[0;32mI (21680) monitor:           telnet (2)  0% s: 3200␛[0m
␛[0;32mI (31720) monitor: Heap internal:137271 (min:129043) external:3994623 (min:3965911) dma:91559 (min:83347)␛[0m
␛[0;32mI (31720) monitor:             IDLE (1) 99% s:  948     pseudo_idle (0)  0% s: 3316            IDLE (1) 97% s:  948␛[0m
␛[0;32mI (31740) monitor:          Tmr Svc (2)  0% s: 1200             tiT (2)  0% s: 1884            ipc1 (2)  0% s:  464␛[0m
␛[0;32mI (31750) monitor:             ipc0 (2)  0% s:  468         sys_evt (2)  0% s:  808            mdns (2)  0% s: 2972␛[0m
␛[0;32mI (31760) monitor:          buttons (2)  0% s: 1548         network (2)  0% s:  800           httpd (2)  0% s: 1140␛[0m
␛[0;32mI (31780) monitor:             wifi (2)  1% s: 3196       esp_timer (3)  0% s: 3372          telnet (2)  0% s: 3200␛[0m
␛[0;32mI (31800) monitor:          console (2)  0% s: 2828␛[0m
␛[0;32mI (41880) monitor: Heap internal:137271 (min:129043) external:3994623 (min:3965911) dma:91559 (min:83347)␛[0m
␛[0;32mI (41880) monitor:             IDLE (1) 99% s:  948     pseudo_idle (0)  0% s: 3316            IDLE (1) 97% s:  948␛[0m
␛[0;32mI (41920) monitor:          Tmr Svc (2)  0% s: 1200             tiT (2)  0% s: 1884            ipc0 (2)  0% s:  468␛[0m
␛[0;32mI (41920) monitor:          sys_evt (2)  0% s:  808         buttons (2)  0% s: 1548          telnet (2)  0% s: 3200␛[0m
␛[0;32mI (41940) monitor:          network (2)  0% s:  800           httpd (2)  0% s: 1140            mdns (2)  0% s: 2972␛[0m
␛[0;32mI (41960) monitor:             wifi (2)  1% s: 3196       esp_timer (3)  0% s: 3372         console (2)  0% s: 2828␛[0m
␛[0;32mI (41980) monitor:             ipc1 (2)  0% s:  464␛[0m
␛[0;32mI (52020) monitor: Heap internal:137271 (min:129043) external:3994623 (min:3965911) dma:91559 (min:83347)␛[0m
␛[0;32mI (52020) monitor:             IDLE (1) 99% s:  948     pseudo_idle (0)  0% s: 3316            IDLE (1) 97% s:  948␛[0m
␛[0;32mI (52040) monitor:          Tmr Svc (2)  0% s: 1200             tiT (2)  0% s: 1884         sys_evt (2)  0% s:  808␛[0m
␛[0;32mI (52060) monitor:          buttons (2)  0% s: 1548          telnet (2)  0% s: 3200         console (2)  0% s: 2828␛[0m
␛[0;32mI (52060) monitor:             mdns (2)  0% s: 2972         network (2)  0% s:  800           httpd (2)  0% s: 1140␛[0m
␛[0;32mI (52080) monitor:             wifi (2)  1% s: 3196       esp_timer (3)  0% s: 3372            ipc1 (2)  0% s:  464␛[0m
␛[0;32mI (52100) monitor:             ipc0 (2)  0% s:  468␛[0m

**Comment by philippe44** [comment]
Oh, you are still in recovery mode. You need to exit it using the webui or "restart_ota" in the command line.

**Reply by DonKracho** [reply, marked as answer]
Thanks a lot! After that I had to remove the -d option in the squeezelite command line, then squeezelite starts and the player is shown in LMS. It connects to the BT Speaker too. But the BT output ist distorted, sounds like a sampling rate issue. I tried Radios, to ensure to have a low res source. But nevertheless good progress here. Now it is running basically. Great project, thanks to all contributors. I will read through the discussions to get the BT issue sorted out.

**Reply by DonKracho** [reply]
OTA flashed the latest 16bit I2S release. The output to a BT-Speaker is working fine now👌👍

**Comment by DonKracho** [comment]
Just two links which helped me a lot getting all the stuff running fine:
https://squeezeesp32.blogspot.com/p/beginners-guide.html

With this clone module I faced an issue that it got stuck in a boot loop after a power cycle. You will get help here:
https://github.com/sle118/squeezelite-esp32/issues/270
